baseline_xgb/imbalanceness.py

```diff
@@ -0,0 +1,26 @@
+import pandas as pd
+import numpy as np
+
+# read in the dataset
+dataset = pd.read_csv("/gpfs/data/oermannlab/project_data/text2table/complete_v2/dataset_v2.csv")
+
+# empirical entropy
+def emp_entropy(series):
+    # dummify the column on which we are calculating empirical entropy
+    dummified = series.astype(str).dropna().str.get_dummies(sep=" <CEL> ")
+    # count the number of samples for every unique combinations of labels
+    unique_combs = dummified.groupby(list(dummified.columns)).size().reset_index().rename(columns={0: "count"})
+    count = unique_combs["count"].astype(float)
+    # calculate the entropy
+    entropy = (count/count.sum())*np.log2(count/count.sum())
```

Comment:
This script is for the empirical entropy of each task instead of the entropy of each label in the tasks. Computation of entropy of each label is incorporated in new_xgb.py.

Reply:
please take a look at line 110 to line 116 of new_xgb.py

Reply:
I kept this script in case we need it at some time in the future.